### ISSUE TITLE
deploy:shared task fails when using multi-level directory in shared_dirs

### ISSUE THREAD

**Issue by kirkdocker** [deployphp/deployer#916]
| Q                 | A
| ----------------- | ---
| Issue Type        | Bug
| Deployer Version  | 4.0.1
| Local Machine OS  | Windows (git bash)
| Remote Machine OS | Ubuntu (vagrant)

### Description 
When setting the shared_dirs config option, if you use a multi-level directory such as 'laravel/storage' the deploy:shared task fails to copy and link correctly.

### Steps to reproduce

Create an app which has a laravel folder and storage folder within that, with a few files/folders in there.

Set shared_dirs as follows:

```
set('shared_dirs', [
    'laravel/storage'
]);
```

Run the deploy. The following will occur:

In shared directory: there will be the following folders:

laravel/storage (which is empty)
storage (contains contents of releases/1/laravel/storage)

In the release folder: 

laravel/storage (this is a symlink to shared/laravel/storage (which is empty))

### Content of `deploy.php`
```php
<?php
namespace Deployer;
require 'recipe/common.php';

set('repository', 'git@REPO');

set('use_relative_symlink', false);

set('shared_dirs', [
    'laravel/storage'
]);

server('vagrant', '127.0.0.1', '2222')
    ->user('vagrant')
    ->identityFile("", 'keyfile')
    ->set('branch', "'#575_deployment'")
    ->stage('local')
    ->set('deploy_path', '/var/www/local.SITE.com');

desc('Deploy your project');
task('deploy', [
    'deploy:prepare',
    'deploy:lock',
    'deploy:release',
    'deploy:update_code',
    'deploy:shared'
]);
after('deploy', 'success');
```

### Output log
```
Ô×ñ Executing task deploy:shared
[vagrant] > mkdir -p /var/www/local.SITE.com/shared/laravel/storage
[vagrant] > if [ -d $(echo /var/www/local.SITE.com/releases/1/laravel/storage) ]; then cp -rn /var/www/local.SITE.com/releases/1/laravel/storage /var/www/local.SITE.com/shared; fi
[vagrant] > if [ -d $(echo /var/www/local.SITE.com/releases/1/laravel/storage) ]; then rm -rf /var/www/local.SITE.com/releases/1/laravel/storage; fi
[vagrant] > mkdir -p `dirname /var/www/local.SITE.com/releases/1/laravel/storage`
[vagrant] > ln -nfs /var/www/local.SITE.com/shared/laravel/storage /var/www/local.SITE.com/releases/1/laravel/storage
ÔÇó done on [vagrant]
Ô£ö Ok [77ms]
Ô×ñ Executing task success
Ô£ö Ok [1ms]
Successfully deployed!
```

**Comment by kirkdocker** [comment]
Editing the following seems to fix the issue, and still works for single level directories. I've only done limited testing though:

recipe/deploy/shared.php:19
```
run("if [ -d $(echo {{release_path}}/$dir) ]; then cp -rn {{release_path}}/$dir $sharedPath; fi");
```

change to 

```
run("if [ -d $(echo {{release_path}}/$dir) ]; then cp -rn {{release_path}}/$dir/* $sharedPath/$dir; fi");
```

Let me know your thoughts.

Thanks,

**Comment by pdbreen** [comment]
Should be fixed in v4.0.2

**Comment by elfet** [comment]
Fixed in 4.0.2.

**Comment by kirkdocker** [comment]
Ah, awesome, thanks guys.
